narwhal/network/src/connectivity.rs

```diff
@@ -59,40 +70,58 @@ impl ConnectionMonitor {
 
         // now report the connected peers
         for peer_id in connected_peers {
-            self.handle_peer_connect(peer_id);
+            self.handle_peer_connect(peer_id).await;
         }
 
         while let Ok(event) = subscriber.recv().await {
             match event {
                 anemo::types::PeerEvent::NewPeer(peer_id) => {
-                    self.handle_peer_connect(peer_id);
+                    self.handle_peer_connect(peer_id).await;
                 }
                 anemo::types::PeerEvent::LostPeer(peer_id, _) => {
-                    self.handle_peer_disconnect(peer_id);
+                    self.handle_peer_disconnect(peer_id).await;
                 }
             }
         }
     }
 
-    fn handle_peer_connect(&self, peer_id: PeerId) {
+    async fn handle_peer_connect(&self, peer_id: PeerId) {
         self.connection_metrics.network_peers.inc();
 
         if let Some(ty) = self.peer_id_types.get(&peer_id) {
             self.connection_metrics
                 .network_peer_connected
                 .with_label_values(&[&format!("{peer_id}"), ty])
-                .set(1)
+                .set(1);
+
+            match &self.sender {
+                Some(s) => {
+                    if let Err(e) = s.send((peer_id, ConnectionStatus::Connected)).await {
+                        error!("Error sending connection status {e}");
+                    }
+                }
+                None => {}
+            }
```

Comment:
You probably want to put this outside of the outer `if` statement, otherwise we'll attempt to use the `sender` only if `peer_id` is found in the `peer_id_types` in the `map` , which might not be the case (ex we can totally pass in an empty `peer_id_types` map because we don't have tags to associate with , but we might still want to propagate the connectivity status)